adapters/taboola/taboola.go

```diff
@@ -106,12 +96,23 @@ func (a *adapter) MakeBids(request *openrtb2.BidRequest, requestData *adapters.R
 	return bidResponse, errs
 }
 
-func (a *adapter) buildRequest(request *openrtb2.BidRequest, mediaType string) (*adapters.RequestData, error) {
+func (a *adapter) buildRequest(request *openrtb2.BidRequest) (*adapters.RequestData, error) {
 	requestJSON, err := json.Marshal(request)
 	if err != nil {
 		return nil, err
 	}
 
+	//set MediaType based on first imp
+	var mediaType string
+	if request.Imp[0].Banner != nil {
+		mediaType = "display"
+	} else if request.Imp[0].Native != nil {
+		mediaType = "native"
+	} else {
+		fmt.Errorf("unsupported media type for imp: %v", request.Imp[0])
+		return nil, err
```

Comment:
This code will probably will not be reachable since core framework already checks for Bid type based on `.yaml` file. However, if you would like to include it then this line is potential bug since err = nil always instead I would suggest either to remove this else or do - 
```
return nil, fmt.Errorf("unsupported media type for imp: %v", request.Imp[0])
```

Reply:
Agree, L112 is not returned or assigned to a variable.

Reply:
@gargcreation1992  @VeronikaSolovei9 

thanks, did the changes. 

Please will appreciate if you merge this into the next version, I updated the documentation 

https://github.com/prebid/prebid.github.io/pull/4329